protocols/gossipsub/src/behaviour.rs

```diff
@@ -2804,6 +2830,9 @@ where
 
         // Add the IP to the peer scoring system
         if let Some((peer_score, ..)) = &mut self.peer_score {
+            // mxinden: The connection might run via a relay (see relay circuit spec). In that case
```

Comment:
Yeah. I think this is a point of contention in the 1.1 specs. 

There is a scoring configuration parameter that allows you to adjust the number of peers allowed under a single IP before the scoring kicks in. Also you can easily turn this off with the scoring parameters. See the `ip_colocation_factor` parameters in `peer_score/params.rs`. 

Ultimately its an optional scoring param that a user can decide to use or not (and suggested by the specs) and I guess depending on the use of this, is up to the user to include or not.